src/main/scala/util/ShiftReg.scala

```diff
@@ -9,3 +9,43 @@ object ShiftRegisterInit {
       case (next, _) => Reg(next, next = next, init = init)
     }
 }
+
+object ShiftRegister
```

Comment:
can we call this file ShiftRegister since that is the name of the object it creates? Also can you add a comment here that this is functionally the same as Chisel ShiftRegister other than the fact that the registers are named?